modbus_south.cpp

```diff
@@ -40,35 +92,185 @@ Modbus::Modbus(const string& ip, const unsigned short port) :
 	{
 		Logger::getLogger()->info("Modbus TCP connected %s:%d", m_address.c_str(), m_port);
 	}
-}
-
-/**
- * Constructor for the modbus interface for a serial connection
- */
-Modbus::Modbus(const string& device, int baud, char parity, int bits, int stopBits) :
-	m_device(device), m_address(""), m_port(0), m_tcp(false)
-{
-	if ((m_modbus = modbus_new_rtu(device.c_str(), baud, parity, bits, stopBits)) == NULL)
+	if (m_tcp)
```

Comment:
You probably meant "if (!m_tcp)" here? And errno can be reset for this RTU case also before calling "modbus_connect".

Reply:
Yes, something has clearly gone wrong with a copy paste in that bit of code. Have re-orgnaised it.